src/ide_cdrom.cpp

```diff
@@ -1519,23 +1514,30 @@ bool IDECDROMDevice::loadAndValidateCueSheet(FsFile *dir, const char *cuesheetna
         return false;
     }
 
-    if (cuesheetfile.size() > sizeof(m_cuesheet))
+    if (cuesheetfile.size() > SharedCUEParser::max_cue_sheet_size())
     {
         logmsg("---- WARNING: CUE sheet length ", (int)cuesheetfile.size(), " exceeds maximum ",
-                (int)sizeof(m_cuesheet), " bytes");
+                (int)SharedCUEParser::max_cue_sheet_size(), " bytes for device reads and playback");
+        return false;
     }
 
-    int len = cuesheetfile.read(m_cuesheet, sizeof(m_cuesheet));
-    cuesheetfile.close();
-    if (len <= 0)
+    char cuesheetpath[MAX_FILE_PATH + 1];
+    char tmp[MAX_FILE_PATH + 1];
+    dir->getName(tmp, sizeof(tmp));
+    if (strlen(tmp) > 0)
     {
-        m_cuesheet[0] = 0;
-        logmsg("---- Failed to read cue sheet from ", cuesheetname);
-        return false;
+        strcpy(cuesheetpath, "/");
+        strcat(cuesheetpath, tmp);
+        strcat(cuesheetpath, "/");
+    }
+    else
+    {
+        cuesheetpath[0] = '\0';
     }
 
-    m_cuesheet[len] = 0;
-    m_cueparser = CUEParser(m_cuesheet);
+    cuesheetfile.getName(tmp, sizeof(tmp));
+    strcat(cuesheetpath, tmp);
```

Comment:
There is risk of buffer overflow here. The change to give `FsFile` to constructor would avoid having to build the path here.